src/Inputs/scene.inputManager.ts

```diff
@@ -804,64 +806,81 @@ export class InputManager {
+            if (device.deviceType == DeviceType.Keyboard) {
+                this._deviceSourceManager.getDeviceSource(device.deviceType, device.deviceSlot)!.onInputChangedObservable.add((deviceSource) => {
+                    if (elementToAttachTo && document.activeElement === elementToAttachTo) {
+                        // Get Keyboard key and create event init data
+                        const key = String.fromCharCode(deviceSource.inputIndex);
+                        const init = {key: key, keyCode: deviceSource.inputIndex};
+
+                        if (deviceSource.currentState == 1) {
+                            const evt = new KeyboardEvent("keydown", init);
+                            this._onKeyDown(evt);
+                        }
+
+                        if (deviceSource.currentState == 0) {
+                            const evt = new KeyboardEvent("keyup", init);
+                            this._onKeyUp(evt);
+                        }
+                    }
+                });
+            }
+
+            // Pointer Events
+            if (device.deviceType == DeviceType.Mouse || device.deviceType == DeviceType.Touch) {
+                this._deviceSourceManager.getDeviceSource(device.deviceType, device.deviceSlot)!.onInputChangedObservable.add((deviceSource) => {
+                    // Get current cursor position and create event init data
+                    const pointerX = this._deviceSourceManager.getDeviceSource(device.deviceType)!.getInput(PointerInput.Horizontal) || 0;
+                    const pointerY = this._deviceSourceManager.getDeviceSource(device.deviceType)!.getInput(PointerInput.Vertical) || 0;
+                    // If dealing with a change to the delta, grab values for event init
+                    const deltaX = (deviceSource.inputIndex == PointerInput.DeltaHorizontal) ? this._deviceSourceManager.getDeviceSource(device.deviceType)!.getInput(PointerInput.DeltaHorizontal) : 0;
+                    const deltaY = (deviceSource.inputIndex == PointerInput.DeltaVertical) ? this._deviceSourceManager.getDeviceSource(device.deviceType)!.getInput(PointerInput.DeltaVertical) : 0;
+                    const init = {
+                        pointerId: 1,
+                        clientX: pointerX,
+                        clientY: pointerY,
+                        movementX: deltaX,
+                        movementY: deltaY
+                    };
+                    const evt = (device.deviceType == DeviceType.Mouse) ? new PointerEvent("mouse", init) : new PointerEvent("touch", init);
+                    Object.defineProperty(evt, 'target', {value: elementToAttachTo});
 
-        // Keyboard events
-        this._onCanvasFocusObserver = engine.onCanvasFocusObservable.add(
-            (() => {
+                    if (attachDown && deviceSource.inputIndex >= PointerInput.LeftClick && deviceSource.inputIndex <= PointerInput.RightClick && deviceSource.currentState == 1) {   // Pointer Down

```

Comment:
well this seems wrong as this only need to happen when the canvas is getting focused